hydra-node/test/Hydra/ClientInputSpec.hs

```diff
@@ -23,6 +24,33 @@ spec = parallel $ do
     settings
     (Proxy @(ReasonablySized (ClientInput Tx)))
 
+  describe "FromJSON (ValidatedTx era)" $ do
```

Comment:
We do have roundtrip tests on serialization of cardano transactions in `CardanoSpec`. IMO these tests would fit there better than in here. After all `ClientInput` has nothing to do with cardano transactions.

Reply:
> After all `ClientInput` has nothing to do with cardano transactions.

I... kindly disagree :sweat_smile: ? The `ClientInput` is the only place where this serialization is used (`NewTx`)! The `CardanoSpec` regards more the internal representation of various Cardano types, whereas the `ClientInputSpec` relates to our user-facing interface; There are round-trip tests just above but since the `ToJSON` instance will still produce JSON in the "detailed" form, they are no good for testing the alternative parsers.